### PR TITLE
feat: deprecate enforcesemver2 config option

### PR DESCRIPTION
**Pull request by cbuto**
This is a follow up on #482 to fully "deprecate" the `--enforce-semver2` flag. It seems like some of the documentation work is capture in https://github.com/helm/chartmuseum/issues/485, which I can follow up on. 

Also, do we want a way to mark config options as deprecated and print warnings if they are set on startup?

### CODE REVIEW

File: pkg/chartmuseum/server/multitenant/api.go [hunk 1]
```diff
@@ -96,9 +95,20 @@ func (server *MultiTenantServer) deleteChartVersion(log cm_logger.LoggingFn, rep
...
+		// left the others fields to be default
+	})
+	if err != nil {
+		return filename, &HTTPError{http.StatusBadRequest, err.Error()}
+	}
+
+	filename, err = cm_repo.ChartPackageFilenameFromContent(content)
 	if err != nil {
-		// TODO: this is an error if invalid semver, making "EnforceSemver2" useless...
 		return filename, &HTTPError{http.StatusInternalServerError, err.Error()}
```
Comment (cbuto) [inline pkg/chartmuseum/server/multitenant/api.go:112]: While I get familiar with everything, should we return a `400` here and skip the call to `ChartVersionFromStorageObject` entirely?

**Review by jdolitsky — CHANGES_REQUESTED** [review]
> do we want a way to mark config options as deprecated and print warnings
yes, that would be great. Thanks @cbuto!

File: pkg/chartmuseum/server/multitenant/api.go (hunk 1 above)
Comment (cbuto) [reply pkg/chartmuseum/server/multitenant/api.go]: cool, fixed in 2f11e5029b735bc6a92a9fc566a732dfccfa9c01!

File: pkg/chartmuseum/server.go [hunk 2]
```diff
@@ -137,9 +136,10 @@ func NewServer(options ServerOptions) (Server, error) {
 		UseStatefiles:          options.UseStatefiles,
 		AllowOverwrite:         options.AllowOverwrite,
 		AllowForceOverwrite:    options.AllowForceOverwrite,
-		EnforceSemver2:         options.EnforceSemver2,
-		Version:                options.Version,
-		CacheInterval:          options.CacheInterval,
+		// EnforceSemver was deprecated, see https://github.com/helm/chartmuseum/issues/485 for more info
+		EnforceSemver2: options.EnforceSemver2,
+		Version:        options.Version,
+		CacheInterval:  options.CacheInterval,
```
[the remark this answers is not in the document]
Comment (cbuto) [reply pkg/chartmuseum/server.go]: done in 2f11e5029b735bc6a92a9fc566a732dfccfa9c01 👍

**Review by jdolitsky — APPROVED** [review]
👍

**Comment by scbizu** [comment]
Cool @cbuto , it really do me a fever 🥳

**Comment by scbizu** [comment]
Cool @cbuto , it really do me a favor 🥳

**Pull request merged**